components/servo/lib.rs

```diff
@@ -240,11 +247,228 @@ impl<Window> Servo<Window> where Window: WindowMethods + 'static {
         Servo {
             compositor: compositor,
             constellation_chan: constellation_chan,
+            embedder_receiver: embedder_receiver,
+        }
+    }
+
+    fn handle_window_event(&mut self, event: WindowEvent) {
+        match event {
+            WindowEvent::Idle => {
+            }
+
+            WindowEvent::Refresh => {
+                self.compositor.composite();
+            }
+
+            WindowEvent::Resize(size) => {
+                self.compositor.on_resize_window_event(size);
+            }
+
+            WindowEvent::LoadUrl(top_level_browsing_context_id, url) => {
+                let msg = ConstellationMsg::LoadUrl(top_level_browsing_context_id, url);
```

Comment:
Any recommendation of how we could avoid all these translations from WindowEvents to ConstellationMsg? This is quite annoying.